### PR TITLE
Feat/616 Transaction management

### PR DESCRIPTION
**Pull request by mattrussell36**
- Updates to accomodate changes in smart-contracts-sdk
- Adds TransactionModal component to show a list of transactions
- Add usingPendingTx hook to detect when txs are pending
- Allow loader to render with black squares
- Starts listeners for transactions on staking and vesting contracts

Screen Shot 2022-01-13 at 8 35 50 PM

### CODE REVIEW

File: src/contexts/contracts/contracts-provider.tsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import {
+  Networks,
   VegaClaim,
   VegaErc20Bridge,
   VegaStaking,
   VegaToken,
   VegaVesting,
 } from "@vegaprotocol/smart-contracts-sdk";
+import { ethers } from "@vegaprotocol/smart-contracts-sdk/node_modules/ethers";
```
Comment (dexturr) [inline src/contexts/contracts/contracts-provider.tsx:9]: Hmmm not a massive fan of this, can we make this a peer dependency from the smart contracts SDK?

Comment (dexturr) [reply src/contexts/contracts/contracts-provider.tsx:9]: Or externalize, whichever is more appropriate

Comment (mattrussell36) [reply src/contexts/contracts/contracts-provider.tsx]: Think this was an accidental auto import. Its gone now

**Comment by mattrussell36** [comment]
@campbellssource @dexturr Getting the arguments of the transaction (eg. amount/vega key) to display in the modal is a bit tricky. We could pass it in when we call the contract method and store it but this feels a bit wrong to me... If we pick up the tx from an event listener we won't know what transaction triggered it. We can however get the arguments from the event, but we don't have the event when we trigger the transaction from the app. Hope that makes sense. 

Do we think this its acceptable to just show the tx hash (and link to Etherscan) and the pending status as a first iteration?

File: src/components/transactions-modal/transactions-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+              </thead>
+              <tbody>
+                {transactions.map((t) => {
+                  return (
+                    <tr>
+                      <td>
+                        <EtherscanLink
+                          tx={t.tx.hash}
+                          text={truncateMiddle(t.tx.hash)}
+                          chainId={`0x${chainId}` as any}
```
Comment (dexturr) [inline src/components/transactions-modal/transactions-modal.tsx:95]:
```suggestion
                          chainId={`0x${chainId}` as EthereumChainId}
```

**Review by dexturr — APPROVED** [review]
LGTM, on minor thing, just wondering what we are saying is the new process for unit testing. Should I be annoying you about it in this PR or do we want to start as of the next feature?

**Pull request merged**